packages/tpetra/core/ext/TpetraExt_MatrixMatrix_def.hpp

```diff
@@ -479,6 +478,103 @@ void Add(
   }
 }
 
+namespace ColMapFunctors
+{
+  template<typename ByteView, typename GView>
+  struct UnionEntries
+  {
+    UnionEntries(ByteView& entryUnion_, const GView& gids_) : entryUnion(entryUnion_), gids(gids_) {}
+    KOKKOS_INLINE_FUNCTION void operator()(const size_t i) const
+    {
+      entryUnion(gids(i)) = 1;
+    }
+    ByteView& entryUnion;
```

Comment:
Kokkos::View has shallow copy semantics.  If your functor holds onto Kokkos::View instances by nonconst reference, this functor won't work on GPUs.  A reference is a pointer, and it points to host memory, where the class instance lives.  The class instance _contains_ a GPU device allocation.  Just store the Kokkos::View by value in the functor.